apps/veil/src/entities/leaderboard/ui/table.tsx

```diff
@@ -1,76 +1,84 @@
 'use client';
 
 import { useAutoAnimate } from '@formkit/auto-animate/react';
-import { useState, useMemo, useRef } from 'react';
+import { useState, useRef } from 'react';
 import cn from 'clsx';
-import orderBy from 'lodash/orderBy';
 import Link from 'next/link';
-import { useSearchParams } from 'next/navigation';
+import { useSearchParams, useParams } from 'next/navigation';
 import { SquareArrowOutUpRight } from 'lucide-react';
 import { Text } from '@penumbra-zone/ui/Text';
 import { Card } from '@penumbra-zone/ui/Card';
 import { TableCell } from '@penumbra-zone/ui/TableCell';
 import { SegmentedControl } from '@penumbra-zone/ui/SegmentedControl';
 import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
 import { Pagination } from '@penumbra-zone/ui/Pagination';
-import { AssetSelector, AssetSelectorValue } from '@penumbra-zone/ui/AssetSelector';
-import { useAssets } from '@/shared/api/assets';
-import { useBalances } from '@/shared/api/balances';
-import { useLeaderboard } from '@/entities/leaderboard/api/use-leaderboard';
 import { stateToString } from '@/entities/position/model/state-to-string';
 import { useSortableTableHeaders } from '@/pages/tournament/ui/sortable-table-header';
-import { formatAge, getAssetId } from './utils';
-import { useTournamentSummary } from '@/pages/tournament/api/use-tournament-summary';
+import { useLpLeaderboard } from '@/entities/leaderboard/api/use-lp-leaderboard';
+import { LpLeaderboardSortKey } from '@/entities/leaderboard/api/utils';
+import { pnum } from '@penumbra-zone/types/pnum';
+import { observer } from 'mobx-react-lite';
+import { connectionStore } from '@/shared/model/connection';
+import { useMyLpLeaderboard } from '@/entities/leaderboard/api/use-my-lp-leaderboard';
+import { round } from '@penumbra-zone/types/round';
+import { useStakingTokenMetadata } from '@/shared/api/registry';
+import { useGetMetadata } from '@/shared/api/assets';
 
-export const LeaderboardTable = () => {
-  const totalCountRef = useRef<number>(0);
+const Tabs = {
+  AllLPs: 'All LPs',
+  MyLPs: 'My LPs',
+} as const;
+
+type Tab = (typeof Tabs)[keyof typeof Tabs];
+
+export const LeaderboardTable = observer(() => {
+  const { connected, subaccount } = connectionStore;
+  const totalRef = useRef<number>(0);
+  const params = useParams<{ epoch: string }>();
+  const epoch = Number(params?.epoch);
```

Comment:
suggestion (consistency): accept epoch as a prop to align with another components of the round page